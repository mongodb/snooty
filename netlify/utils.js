import fs from 'fs';
import path from 'path';

/**
 * Parses build hook data from the expected temporary location
 * @returns {object | undefined}
 */
const parseBuildHookData = () => {
  // This file does not currently exist, but should be created on Netlify builds as part of `npm run build:netlify`.
  // The INCOMING_HOOK_BODY env var is not automatically passed along to functions, so we use a txt file to save it
  const relativeFilePath = '../build-hook.txt';
  const buildHookDataString = fs.readFileSync(path.resolve(__dirname, relativeFilePath), 'utf-8');
  if (!buildHookDataString) {
    console.log('No build hook data found.');
    return;
  }
  return JSON.parse(buildHookDataString);
};

export const constructResPayload = (event) => {
  const buildHookData = parseBuildHookData();
  const parsedEventBody = JSON.parse(event.body);
  // This is Netlify's default post-deployment payload. We include it with our custom data in case
  // we want to process any information
  const netlifyPayload = parsedEventBody.payload;
  return {
    netlifyPayload,
    ...buildHookData,
  };
};
