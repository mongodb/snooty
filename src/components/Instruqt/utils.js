export const getLabURL = (embedValue) => {
  return `https://play.instruqt.com/embed${embedValue}`;
};
