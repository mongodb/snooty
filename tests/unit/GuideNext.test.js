import React from 'react';
import { render } from '@testing-library/react';
import mockChaptersData from './data/Chapters.test.json';
import mockNodeData from './data/GuideNext.test.json';
import GuideNext from '../../src/components/GuideNext/GuideNext';
import * as browserStorage from '../../src/utils/browser-storage';

const renderGuideNext = (slug, mockNodeData = {}) => {
  return render(<GuideNext nodeData={mockNodeData} metadata={mockChaptersData.metadata} slug={slug} />);
};

describe('GuideNext', () => {
  jest.useFakeTimers();
  const mockedLocalStorage = {};
  let mockedSetLocalValue;

  beforeAll(() => {
    // Mock browser storage to make sure that this component is properly saving the values
    mockedSetLocalValue = jest
      .spyOn(browserStorage, 'setLocalValue')
      .mockImplementation((localStorageKey, readGuides) => {
        mockedLocalStorage[localStorageKey] = readGuides;
      });
  });

  afterAll(() => {
    mockedSetLocalValue.mockClear();
  });

  const checkLocalStorage = (slug) => {
    expect(mockedLocalStorage['readGuides'][slug]).toBeTruthy();
  };

  it('renders the next guide in the same chapter', () => {
    const currentSlug = 'cloud/account';
    const wrapper = renderGuideNext(currentSlug, mockNodeData.noContent);
    expect(wrapper.asFragment()).toMatchSnapshot();
    checkLocalStorage(currentSlug);
  });

  it('renders the first guide in the next chapter', () => {
    const currentSlug = 'cloud/migrate-from-aws-to-atlas';
    const wrapper = renderGuideNext(currentSlug, mockNodeData.noContent);
    expect(wrapper.asFragment()).toMatchSnapshot();
    checkLocalStorage(currentSlug);
  });

  it('renders the default copy on the final guide', () => {
    const currentSlug = 'server/read';
    const wrapper = renderGuideNext(currentSlug, mockNodeData.noContent);
    expect(wrapper.asFragment()).toMatchSnapshot();
    checkLocalStorage(currentSlug);
  });

  it('renders custom copy', () => {
    const currentSlug = 'server/read';
    const wrapper = renderGuideNext(currentSlug, mockNodeData.customContent);
    expect(wrapper.getByText('Custom title with content and button')).toBeTruthy();
    expect(wrapper.getByText('Hello this is my custom content. Description here!')).toBeTruthy();
    // Default copy title should not be present
    expect(wrapper.queryAllByText('Become a MongoDB Professional')).toHaveLength(0);
  });
});
