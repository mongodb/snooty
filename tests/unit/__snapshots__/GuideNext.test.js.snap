// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideNext renders the default copy on the final guide 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-top: 1px solid #E7EEEC;
  margin-bottom: 24px;
  padding-top: 56px;
}

@media not all and (max-width: 1023px) {
  .emotion-0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 34px;
  }
}

.emotion-2 p {
  margin-top: 8px;
}

@media not all and (max-width: 1023px) {
  .emotion-2 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-4 {
  color: #5D6C74;
  font-size: 16px;
  margin-bottom: 16px;
}

.emotion-6 {
  font-size: 24px;
  font-weight: bold;
}

.emotion-8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #00684A;
  border-color: #00684A;
  color: #FFFFFF;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
}

.emotion-8:focus {
  outline: none;
}

.emotion-8[disabled],
.emotion-8:disabled {
  pointer-events: none;
}

.emotion-8:active,
.emotion-8:focus,
.emotion-8:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-8:hover,
.emotion-8:active {
  color: #FFFFFF;
  background-color: #00593f;
  border-color: #00593f;
  box-shadow: 0 0 0 3px #C0FAE6;
}

.emotion-8:focus {
  color: #FFFFFF;
  background-color: #00593f;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #007CAD;
  cursor: pointer;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.emotion-9:focus {
  outline: none;
}

.emotion-10 {
  background-repeat: repeat-x;
  background-size: 2px 2px;
  background-position: center bottom;
}

[data-leafygreen-ui="anchor-container"]:hover .emotion-10 {
  background-image: linear-gradient( #E7EEEC 100%, #E7EEEC 0 );
}

[data-leafygreen-ui="anchor-container"]:focus .emotion-10 {
  background-image: linear-gradient( to right, #9DD0E7 100%, #9DD0E7 0 );
}

.emotion-10 {
  background-repeat: repeat-x;
  -webkit-background-size: 2px 2px;
  background-size: 2px 2px;
  -webkit-background-position: center bottom;
  background-position: center bottom;
}

[data-leafygreen-ui="anchor-container"]:hover .emotion-10 {
  background-image: linear-gradient(
        #E7EEEC 100%,
        #E7EEEC 0
      );
}

[data-leafygreen-ui="anchor-container"]:focus .emotion-10 {
  background-image: linear-gradient(
        to right,
        #9DD0E7 100%,
        #9DD0E7 0
      );
}

.emotion-11 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-12 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-13 {
  margin-top: 40px;
  padding: 40px 24px;
}

@media not all and (max-width: 767px) {
  .emotion-13 {
    padding: 48px;
  }
}

@media not all and (max-width: 1023px) {
  .emotion-13 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    min-width: 300px;
    margin-top: 0;
    max-width: 510px;
  }
}

.emotion-14 {
  position: relative;
  border-radius: 7px;
  -webkit-transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  border: 1px solid #E7EEEC;
  box-shadow: 0 4px 10px -4px rgba(6,22,33,0.3);
  background-color: white;
  color: #21313C;
}

.emotion-15 {
  margin-bottom: 24px;
}

@media not all and (max-width: 767px) {
  .emotion-15 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 16px;
    margin-bottom: 34px;
  }
}

.emotion-18 {
  background-color: #E4F4E4;
  border-radius: 4px;
  color: #13AA52;
  font-size: 14px;
  font-weight: bold;
  height: 24px;
  text-align: center;
  width: 83px;
  margin-bottom: 8px;
}

.emotion-20 {
  color: #061621;
  font-weight: bold;
}

.emotion-22 {
  list-style: none;
  padding-left: 0;
}

.emotion-24 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 40px;
  padding-top: 0!important;
  position: relative;
}

.emotion-24:not(:last-child) {
  padding-bottom: 16px;
}

.emotion-24:not(:last-child):before {
  content: '';
  position: absolute;
  left: 10px;
  top: 20px;
  height: calc(100% - 20px);
  border-left: solid 1px #13AA52;
}

.emotion-26 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border: solid 1px transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 12px;
  min-width: 20px;
  min-height: 20px;
}

.emotion-28 {
  color: #13AA52;
}

.emotion-29 {
  color: #061621;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-29:hover,
.emotion-29:active {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        What's Next
      </div>
      <div
        class="emotion-6 emotion-7"
      >
        Become a MongoDB Professional
      </div>
      <p>
        Congrats. Youâ€™ve completed all the guides. Want to take the next step? Register for the developer exam.
      </p>
      <a
        aria-disabled="false"
        class="emotion-8 emotion-9"
        data-leafygreen-ui="button"
        href="https://university.mongodb.com/certification/developer/about"
        rel="noopener noreferrer"
        target="_blank"
        type="button"
      >
        <span
          class="emotion-10"
        >
          <div
            class="emotion-11"
          />
          <div
            class="emotion-12"
          >
            Learn More
          </div>
        </span>
      </a>
    </div>
    <div
      class="emotion-13 emotion-14"
    >
      <div
        class="emotion-15 emotion-16"
      >
        <div
          class="emotion-17 emotion-18 emotion-19"
        >
          Chapter 2
        </div>
        <div
          class="emotion-20 emotion-21"
        >
          CRUD
        </div>
      </div>
      <ul
        class="emotion-22 emotion-23"
      >
        <li
          class="emotion-24 emotion-25"
          color="#13AA52"
        >
          <div
            class="emotion-26 emotion-27"
            color="transparent"
          >
            <svg
              aria-label="Checkmark Icon"
              class="emotion-28"
              fill="none"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.306 9.05l5.455-5.455a1 1 0 011.414 0l.707.707a1 1 0 010 1.414l-7.067 7.068a1 1 0 01-1.5-.098l-3.049-3.97a1 1 0 01.184-1.402l.595-.457a1.25 1.25 0 011.753.23L6.306 9.05z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <a
            class="emotion-29 emotion-30"
            href="/server/read/"
          >
            Read Data in MongoDB
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GuideNext renders the first guide in the next chapter 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-top: 1px solid #E7EEEC;
  margin-bottom: 24px;
  padding-top: 56px;
}

@media not all and (max-width: 1023px) {
  .emotion-0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 34px;
  }
}

.emotion-2 p {
  margin-top: 8px;
}

@media not all and (max-width: 1023px) {
  .emotion-2 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-4 {
  color: #5D6C74;
  font-size: 16px;
  margin-bottom: 16px;
}

.emotion-6 {
  font-size: 24px;
  font-weight: bold;
}

.emotion-8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #00684A;
  border-color: #00684A;
  color: #FFFFFF;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
}

.emotion-8:focus {
  outline: none;
}

.emotion-8[disabled],
.emotion-8:disabled {
  pointer-events: none;
}

.emotion-8:active,
.emotion-8:focus,
.emotion-8:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-8:hover,
.emotion-8:active {
  color: #FFFFFF;
  background-color: #00593f;
  border-color: #00593f;
  box-shadow: 0 0 0 3px #C0FAE6;
}

.emotion-8:focus {
  color: #FFFFFF;
  background-color: #00593f;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-9 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-10 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-11 {
  margin-top: 40px;
  padding: 40px 24px;
}

@media not all and (max-width: 767px) {
  .emotion-11 {
    padding: 48px;
  }
}

@media not all and (max-width: 1023px) {
  .emotion-11 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    min-width: 300px;
    margin-top: 0;
    max-width: 510px;
  }
}

.emotion-12 {
  position: relative;
  border-radius: 7px;
  -webkit-transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  border: 1px solid #E7EEEC;
  box-shadow: 0 4px 10px -4px rgba(6,22,33,0.3);
  background-color: white;
  color: #21313C;
}

.emotion-13 {
  margin-bottom: 24px;
}

@media not all and (max-width: 767px) {
  .emotion-13 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 16px;
    margin-bottom: 34px;
  }
}

.emotion-16 {
  background-color: #E4F4E4;
  border-radius: 4px;
  color: #13AA52;
  font-size: 14px;
  font-weight: bold;
  height: 24px;
  text-align: center;
  width: 83px;
  margin-bottom: 8px;
}

.emotion-18 {
  color: #061621;
  font-weight: bold;
}

.emotion-20 {
  list-style: none;
  padding-left: 0;
}

.emotion-22 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 40px;
  padding-top: 0!important;
  position: relative;
}

.emotion-22:not(:last-child) {
  padding-bottom: 16px;
}

.emotion-22:not(:last-child):before {
  content: '';
  position: absolute;
  left: 10px;
  top: 20px;
  height: calc(100% - 20px);
  border-left: solid 1px #E7EEEC;
}

.emotion-24 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border: solid 1px #13AA52;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 12px;
  min-width: 20px;
  min-height: 20px;
}

.emotion-26 {
  color: #13AA52;
}

.emotion-27 {
  color: #061621;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-27:hover,
.emotion-27:active {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        What's Next
      </div>
      <div
        class="emotion-6 emotion-7"
      >
        Read Data in MongoDB
      </div>
      <a
        aria-disabled="false"
        class="emotion-8"
        data-leafygreen-ui="button"
        href="/server/read/"
        type="button"
      >
        <div
          class="emotion-9"
        />
        <div
          class="emotion-10"
        >
          Start Guide
        </div>
      </a>
    </div>
    <div
      class="emotion-11 emotion-12"
    >
      <div
        class="emotion-13 emotion-14"
      >
        <div
          class="emotion-15 emotion-16 emotion-17"
        >
          Chapter 2
        </div>
        <div
          class="emotion-18 emotion-19"
        >
          CRUD
        </div>
      </div>
      <ul
        class="emotion-20 emotion-21"
      >
        <li
          class="emotion-22 emotion-23"
          color="#E7EEEC"
        >
          <div
            class="emotion-24 emotion-25"
            color="#13AA52"
          >
            <svg
              class="emotion-26"
              fill="currentColor"
              height="12"
              viewBox="0 0 7 12"
              width="7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.47516 3.75H3.95081L4.88268 0.707813C4.97017 0.351563 4.71862 0 4.37518 0H1.22522C0.96272 0 0.739598 0.208594 0.704598 0.4875L0.00460634 6.1125C-0.0369557 6.45 0.208042 6.75 0.525225 6.75H3.12176L2.11333 11.3086C2.03458 11.6648 2.28833 12 2.62301 12C2.80676 12 2.98176 11.8969 3.07801 11.7188L6.92797 4.59375C7.1314 4.22109 6.87984 3.75 6.47516 3.75Z"
              />
            </svg>
          </div>
          <a
            class="emotion-27 emotion-28"
            href="/server/read/"
          >
            Read Data in MongoDB
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GuideNext renders the next guide in the same chapter 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-top: 1px solid #E7EEEC;
  margin-bottom: 24px;
  padding-top: 56px;
}

@media not all and (max-width: 1023px) {
  .emotion-0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 34px;
  }
}

.emotion-2 p {
  margin-top: 8px;
}

@media not all and (max-width: 1023px) {
  .emotion-2 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-4 {
  color: #5D6C74;
  font-size: 16px;
  margin-bottom: 16px;
}

.emotion-6 {
  font-size: 24px;
  font-weight: bold;
}

.emotion-8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #00684A;
  border-color: #00684A;
  color: #FFFFFF;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
}

.emotion-8:focus {
  outline: none;
}

.emotion-8[disabled],
.emotion-8:disabled {
  pointer-events: none;
}

.emotion-8:active,
.emotion-8:focus,
.emotion-8:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-8:hover,
.emotion-8:active {
  color: #FFFFFF;
  background-color: #00593f;
  border-color: #00593f;
  box-shadow: 0 0 0 3px #C0FAE6;
}

.emotion-8:focus {
  color: #FFFFFF;
  background-color: #00593f;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-9 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-10 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-11 {
  margin-top: 40px;
  padding: 40px 24px;
}

@media not all and (max-width: 767px) {
  .emotion-11 {
    padding: 48px;
  }
}

@media not all and (max-width: 1023px) {
  .emotion-11 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    min-width: 300px;
    margin-top: 0;
    max-width: 510px;
  }
}

.emotion-12 {
  position: relative;
  border-radius: 7px;
  -webkit-transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  transition: border 300ms ease-in-out,box-shadow 300ms ease-in-out;
  border: 1px solid #E7EEEC;
  box-shadow: 0 4px 10px -4px rgba(6,22,33,0.3);
  background-color: white;
  color: #21313C;
}

.emotion-13 {
  margin-bottom: 24px;
}

@media not all and (max-width: 767px) {
  .emotion-13 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    gap: 0 16px;
    margin-bottom: 34px;
  }
}

.emotion-16 {
  background-color: #E4F4E4;
  border-radius: 4px;
  color: #13AA52;
  font-size: 14px;
  font-weight: bold;
  height: 24px;
  text-align: center;
  width: 83px;
  margin-bottom: 8px;
}

.emotion-18 {
  color: #061621;
  font-weight: bold;
}

.emotion-20 {
  list-style: none;
  padding-left: 0;
}

.emotion-22 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 40px;
  padding-top: 0!important;
  position: relative;
}

.emotion-22:not(:last-child) {
  padding-bottom: 16px;
}

.emotion-22:not(:last-child):before {
  content: '';
  position: absolute;
  left: 10px;
  top: 20px;
  height: calc(100% - 20px);
  border-left: solid 1px #13AA52;
}

.emotion-24 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border: solid 1px transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 12px;
  min-width: 20px;
  min-height: 20px;
}

.emotion-26 {
  color: #13AA52;
}

.emotion-27 {
  color: #061621;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-27:hover,
.emotion-27:active {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-29 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 40px;
  padding-top: 0!important;
  position: relative;
}

.emotion-29:not(:last-child) {
  padding-bottom: 16px;
}

.emotion-29:not(:last-child):before {
  content: '';
  position: absolute;
  left: 10px;
  top: 20px;
  height: calc(100% - 20px);
  border-left: solid 1px #E7EEEC;
}

.emotion-31 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  border: solid 1px #13AA52;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-right: 12px;
  min-width: 20px;
  min-height: 20px;
}

.emotion-33 {
  color: #13AA52;
}

.emotion-40 {
  color: #3D4F58;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        What's Next
      </div>
      <div
        class="emotion-6 emotion-7"
      >
        Set Up Atlas Connectivity
      </div>
      <a
        aria-disabled="false"
        class="emotion-8"
        data-leafygreen-ui="button"
        href="/cloud/connectionstring/"
        type="button"
      >
        <div
          class="emotion-9"
        />
        <div
          class="emotion-10"
        >
          Start Guide
        </div>
      </a>
    </div>
    <div
      class="emotion-11 emotion-12"
    >
      <div
        class="emotion-13 emotion-14"
      >
        <div
          class="emotion-15 emotion-16 emotion-17"
        >
          Chapter 1
        </div>
        <div
          class="emotion-18 emotion-19"
        >
          Atlas
        </div>
      </div>
      <ul
        class="emotion-20 emotion-21"
      >
        <li
          class="emotion-22 emotion-23"
          color="#13AA52"
        >
          <div
            class="emotion-24 emotion-25"
            color="transparent"
          >
            <svg
              aria-label="Checkmark Icon"
              class="emotion-26"
              fill="none"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.306 9.05l5.455-5.455a1 1 0 011.414 0l.707.707a1 1 0 010 1.414l-7.067 7.068a1 1 0 01-1.5-.098l-3.049-3.97a1 1 0 01.184-1.402l.595-.457a1.25 1.25 0 011.753.23L6.306 9.05z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <a
            class="emotion-27 emotion-28"
            href="/cloud/account/"
          >
            Create and Manage a MongoDB Account
          </a>
        </li>
        <li
          class="emotion-29 emotion-23"
          color="#E7EEEC"
        >
          <div
            class="emotion-31 emotion-25"
            color="#13AA52"
          >
            <svg
              class="emotion-33"
              fill="currentColor"
              height="12"
              viewBox="0 0 7 12"
              width="7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.47516 3.75H3.95081L4.88268 0.707813C4.97017 0.351563 4.71862 0 4.37518 0H1.22522C0.96272 0 0.739598 0.208594 0.704598 0.4875L0.00460634 6.1125C-0.0369557 6.45 0.208042 6.75 0.525225 6.75H3.12176L2.11333 11.3086C2.03458 11.6648 2.28833 12 2.62301 12C2.80676 12 2.98176 11.8969 3.07801 11.7188L6.92797 4.59375C7.1314 4.22109 6.87984 3.75 6.47516 3.75Z"
              />
            </svg>
          </div>
          <a
            class="emotion-27 emotion-28"
            href="/cloud/connectionstring/"
          >
            Set Up Atlas Connectivity
          </a>
        </li>
        <li
          class="emotion-29 emotion-23"
          color="#E7EEEC"
        >
          <div
            class="emotion-24 emotion-25"
            color="transparent"
          >
            <svg
              class="emotion-40"
              fill="currentColor"
              height="12"
              viewBox="0 0 7 12"
              width="7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.47516 3.75H3.95081L4.88268 0.707813C4.97017 0.351563 4.71862 0 4.37518 0H1.22522C0.96272 0 0.739598 0.208594 0.704598 0.4875L0.00460634 6.1125C-0.0369557 6.45 0.208042 6.75 0.525225 6.75H3.12176L2.11333 11.3086C2.03458 11.6648 2.28833 12 2.62301 12C2.80676 12 2.98176 11.8969 3.07801 11.7188L6.92797 4.59375C7.1314 4.22109 6.87984 3.75 6.47516 3.75Z"
              />
            </svg>
          </div>
          <a
            class="emotion-27 emotion-28"
            href="/cloud/migrate-from-aws-to-atlas/"
          >
            Migrate a MongoDB Replicate Set from AWS to MongoDB Atlas
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
